### PR TITLE
Disable public IP when network is specified

### PR DESCRIPTION
**Pull request by aksharauke**
Dataflow recommendation is to always disable public IP for the Dataflow workers.
Hence keeping public IPs only for internal testing when there is no network/subnetwork specified.

### CODE REVIEW

File: streaming/streaming.go [hunk 1]
```diff
@@ -445,16 +445,22 @@ func LaunchDataflowJob(ctx context.Context, targetProfile profiles.TargetProfile
 	} else {
 		dataflowHostProjectId = dataflowCfg.HostProjectId
 	}
+
 	dataflowSubnetwork := ""
+
+	// if custom network is not selected, use public IP. Typical internal testing flow
```
Comment (Deep1998) [inline streaming/streaming.go:451]: nit: golang comment style should be full sentences aka start with capital letter and end with full stop.

File: streaming/streaming.go [hunk 2]
```diff
@@ -445,16 +445,22 @@ func LaunchDataflowJob(ctx context.Context, targetProfile profiles.TargetProfile
...
+	workerIpAddressConfig := dataflowpb.WorkerIPAddressConfiguration_WORKER_IP_PUBLIC
+
 	if dataflowCfg.Network != "" {
```
Comment (Deep1998) [inline streaming/streaming.go:454]: and also != nil?

File: streaming/streaming.go [hunk 3]
```diff
@@ -445,16 +445,22 @@ func LaunchDataflowJob(ctx context.Context, targetProfile profiles.TargetProfile
 	} else {
 		dataflowHostProjectId = dataflowCfg.HostProjectId
 	}
+
 	dataflowSubnetwork := ""
+
+	// if custom network is not selected, use public IP. Typical internal testing flow
+	workerIpAddressConfig := dataflowpb.WorkerIPAddressConfiguration_WORKER_IP_PUBLIC
```
Comment (Deep1998) [inline streaming/streaming.go:452]: should this be PUBLIC or UNSPECIFIED

**Review by Deep1998 — APPROVED** [review]

File: streaming/streaming.go (hunk 3 above)
Comment (aksharauke) [reply streaming/streaming.go:452]: PUBLIC. This gives the default behavior that exists when the public IP is not explicitly disabled.

File: streaming/streaming.go (hunk 2 above)
Comment (aksharauke) [reply streaming/streaming.go:454]: This was existing check, tested it with and without specifying VPC. So seems fine as is.

**Review by manitgupta — APPROVED** [review]

**Pull request merged**